pkg/tuple/strings.go

```diff
@@ -90,11 +93,24 @@ func String(rel Relationship) (string, error) {
 		return "", err
 	}
 
-	return StringONR(rel.Resource) + "@" + StringONR(rel.Subject) + caveatString, nil
+	expirationString, err := StringExpiration(rel.OptionalExpiration)
+	if err != nil {
+		return "", err
+	}
+
+	return StringONR(rel.Resource) + "@" + StringONR(rel.Subject) + caveatString + expirationString, nil
+}
+
+func StringExpiration(expiration *time.Time) (string, error) {
+	if expiration == nil {
+		return "", nil
+	}
+
+	return "[expiration:" + expiration.Format(expirationFormat) + "]", nil
```

Comment:
This is future proofing - I suspect we'll need error handling here in the future so I added it now so that we don't need to fix all callsites using it in the future

Reply:
I dislike doing so, because raising an error out of callers in turns means *their* callers need to do so. I much prefer to future proof around allowing an error to be returned